### PR TITLE
congestion: add linear imbalance workload to model

### PR DESCRIPTION
**Pull request by jakmeier**
This workload increases the workload per shard linear to their id. It results in about half the shards being congested and half the shards operating below capacity.

Running it with the current settings gives these results.

```txt
WORKLOAD                 STRATEGY                                 BURNT GAS    TRANSACTIONS FINISHED            MAX QUEUE LEN
Linear Imbalance         No queues                                3999 PGas                    10656                   186116
Linear Imbalance         Global TX stop                           2033 PGas                    38478                     1022
Linear Imbalance         Simple backpressure                      2929 PGas                    53960                     1152
```

Resolves https://github.com/near/nearcore/issues/10705

### CODE REVIEW

**Comment by jakmeier** [comment]
cc @wacban

File: tools/congestion-model/src/workload/linear_imbalance.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+///
+///     receipts_per_round(shard_id) =
+///         2 * (shard_id + 1) * (GAS_LIMIT / GAS_PER_RECEIPT) / NUM_SHARDS
+pub struct LinearImbalanceProducer {
+    pub receipt_size: u64,
+    pub attached_gas: GGas,
+    pub execution_gas: GGas,
+    pub conversion_gas: GGas,
+
+    /// How many message are sent for each directed shard pair (a,b) where a <= b.
```
Comment (wacban) [inline tools/congestion-model/src/workload/linear_imbalance.rs:27]: typo: messages

File: tools/congestion-model/src/workload/linear_imbalance.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+/// still able to function undisturbed.
+///
+/// The work intensity is scaled such that half the shard are below max capacity
+/// and half are above (forced to be congested). And the busiest shard (highest
+/// shard id) receives twice its capacity.
+///
+/// As an actual formula, it looks like this:
+///
+///     receipts_per_round(shard_id) =
+///         2 * (shard_id + 1) * (GAS_LIMIT / GAS_PER_RECEIPT) / NUM_SHARDS
```
Comment (wacban) [inline tools/congestion-model/src/workload/linear_imbalance.rs:20]: 👍

**Review by wacban — APPROVED** [review]
LGTM

File: tools/congestion-model/src/workload/linear_imbalance.rs (hunk 1 above)
Comment (jakmeier) [reply tools/congestion-model/src/workload/linear_imbalance.rs]: thanks, fixed

**Pull request merged**